daemon/api_prompting.go

```diff
@@ -0,0 +1,381 @@
+		return BadRequest("%v", err)
+	}
+
+	return SyncResponse(satisfiedPromptIDs)
+}
+
+func getRules(c *Command, r *http.Request, user *auth.UserState) Response {
+	userID, errorResp := getUserID(r)
+	if errorResp != nil {
+		return errorResp
+	}
+
+	if !getInterfaceManager(c).AppArmorPromptingRunning() {
+		return InternalError("Apparmor Prompting is not running")
+	}
+
+	query := r.URL.Query()
+	snap := query.Get("snap")
+	iface := query.Get("interface")
+
+	rules, err := getInterfaceManager(c).InterfacesRequestsManager().Rules(userID, snap, iface)
+	if err != nil {
+		return InternalError("%v", err)
+	}
+
+	return SyncResponse(rules)
+}
+
+func postRules(c *Command, r *http.Request, user *auth.UserState) Response {
+	userID, errorResp := getUserID(r)
+	if errorResp != nil {
+		return errorResp
+	}
+
+	if !getInterfaceManager(c).AppArmorPromptingRunning() {
+		return InternalError("Apparmor Prompting is not running")
+	}
+
+	var postBody postRulesRequestBody
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&postBody); err != nil {
+		return BadRequest("cannot decode request body for rules endpoint: %v", err)
+	}
+
+	switch postBody.Action {
+	case "add":
+		if postBody.AddRule == nil {
+			return BadRequest(`must include "rule" field in request body when action is "add"`)
+		}
+		newRule, err := getInterfaceManager(c).InterfacesRequestsManager().AddRule(userID, postBody.AddRule.Snap, postBody.AddRule.Interface, postBody.AddRule.Constraints, postBody.AddRule.Outcome, postBody.AddRule.Lifespan, postBody.AddRule.Duration)
+		if errors.Is(err, requestrules.ErrPathPatternConflict) {
+			return Conflict("%v", err)
+		} else if err != nil {
+			return BadRequest("%v", err)
+		}
+		return SyncResponse(newRule)
+	case "remove":
+		if postBody.RemoveSelector == nil {
+			return BadRequest(`must include "selector" field in request body when action is "remove"`)
+		}
+		if postBody.RemoveSelector.Snap == "" && postBody.RemoveSelector.Interface == "" {
+			return BadRequest(`must include "snap" and/or "interface" field in "selector"`)
+		}
+		removedRules, err := getInterfaceManager(c).InterfacesRequestsManager().RemoveRules(userID, postBody.RemoveSelector.Snap, postBody.RemoveSelector.Interface)
+		if err != nil {
+			return InternalError("%v", err)
+		}
+		return SyncResponse(removedRules)
+	default:
+		return BadRequest(`"action" field must be "create" or "remove"`)
+	}
+}
+
+func getRule(c *Command, r *http.Request, user *auth.UserState) Response {
+	vars := muxVars(r)
+	id := vars["id"]
+
+	userID, errorResp := getUserID(r)
+	if errorResp != nil {
+		return errorResp
+	}
+
+	ruleID, err := prompting.IDFromString(id)
+	if err != nil {
+		return BadRequest("%v", err)
+	}
+
+	if !getInterfaceManager(c).AppArmorPromptingRunning() {
+		return InternalError("Apparmor Prompting is not running")
+	}
+
+	rule, err := getInterfaceManager(c).InterfacesRequestsManager().RuleWithID(userID, ruleID)
+	if err != nil {
+		return NotFound("%v", err)
```

Comment:
This error can be a `ErrUserNotAllowed` which should be a 403 perhaps

Reply:
We want it to remain `NotFound` in case we move rules to be stored per-user, like we do with prompts.